### PR TITLE
Add  support of VK_KHR_portability_enumeration

### PR DESCRIPTION
**Pull request by christophe-lunarg**

### CODE REVIEW

File: scripts/genvp.py [hunk 1]
```diff
@@ -553,6 +553,18 @@
...
+            {
+                bool foundPortEnum = false;
+                for (size_t i = 0; i < extensions.size(); ++i) {
+                    if (strcmp(extensions[i], VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0) {
+                        foundPortEnum = true;
+                        break;
+                    }
+                }
+                if (foundPortEnum) {
+                    createInfo.flags = VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```
Comment (charles-lunarg) [inline scripts/genvp.py:565]:
```suggestion
                    createInfo.flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```

File: library/include/vulkan/debug/vulkan_profiles.hpp [hunk 2]
```diff
@@ -8147,6 +8147,18 @@ VPAPI_ATTR VkResult vpCreateInstance(const VpInstanceCreateInfo *pCreateInfo,
...
+            {
+                bool foundPortEnum = false;
+                for (size_t i = 0; i < extensions.size(); ++i) {
+                    if (strcmp(extensions[i], VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0) {
+                        foundPortEnum = true;
+                        break;
+                    }
+                }
+                if (foundPortEnum) {
+                    createInfo.flags = VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```
Comment (charles-lunarg) [inline library/include/vulkan/debug/vulkan_profiles.hpp:8159]:
```suggestion
                    createInfo.flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```

File: library/include/vulkan/vulkan_profiles.hpp [hunk 3]
```diff
@@ -8109,6 +8109,18 @@ VPAPI_ATTR VkResult vpCreateInstance(const VpInstanceCreateInfo *pCreateInfo,
...
+            {
+                bool foundPortEnum = false;
+                for (size_t i = 0; i < extensions.size(); ++i) {
+                    if (strcmp(extensions[i], VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0) {
+                        foundPortEnum = true;
+                        break;
+                    }
+                }
+                if (foundPortEnum) {
+                    createInfo.flags = VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```
Comment (charles-lunarg) [inline library/include/vulkan/vulkan_profiles.hpp:8121]:
```suggestion
                    createInfo.flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```

File: library/source/debug/vulkan_profiles.cpp [hunk 4]
```diff
@@ -7921,6 +7921,18 @@ VPAPI_ATTR VkResult vpCreateInstance(const VpInstanceCreateInfo *pCreateInfo,
...
+            {
+                bool foundPortEnum = false;
+                for (size_t i = 0; i < extensions.size(); ++i) {
+                    if (strcmp(extensions[i], VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0) {
+                        foundPortEnum = true;
+                        break;
+                    }
+                }
+                if (foundPortEnum) {
+                    createInfo.flags = VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```
Comment (charles-lunarg) [inline library/source/debug/vulkan_profiles.cpp:7933]:
```suggestion
                    createInfo.flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```

File: library/source/vulkan_profiles.cpp [hunk 5]
```diff
@@ -7883,6 +7883,18 @@ VPAPI_ATTR VkResult vpCreateInstance(const VpInstanceCreateInfo *pCreateInfo,
...
+            {
+                bool foundPortEnum = false;
+                for (size_t i = 0; i < extensions.size(); ++i) {
+                    if (strcmp(extensions[i], VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0) {
+                        foundPortEnum = true;
+                        break;
+                    }
+                }
+                if (foundPortEnum) {
+                    createInfo.flags = VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```
Comment (charles-lunarg) [inline library/source/vulkan_profiles.cpp:7895]:
```suggestion
                    createInfo.flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
```

**Review by charles-lunarg — COMMENTED** [review]
The code should be consistent whether it uses `|=` for adding the portability flag to instance create info. Currently there are no other flags but its better to be additive than replace the set flags in case other flags are added in the future.

File: library/source/vulkan_profiles.cpp (hunk 5 above)
Comment (christophe-lunarg) [reply library/source/vulkan_profiles.cpp]: This is resolved in the following commit

File: library/source/debug/vulkan_profiles.cpp (hunk 4 above)
Comment (christophe-lunarg) [reply library/source/debug/vulkan_profiles.cpp:7933]: This is resolved in the following commit

File: library/include/vulkan/vulkan_profiles.hpp (hunk 3 above)
Comment (christophe-lunarg) [reply library/include/vulkan/vulkan_profiles.hpp]: This is resolved in the following commit

File: library/include/vulkan/debug/vulkan_profiles.hpp (hunk 2 above)
Comment (christophe-lunarg) [reply library/include/vulkan/debug/vulkan_profiles.hpp:8159]: This is resolved in the following commit

File: scripts/genvp.py (hunk 1 above)
Comment (christophe-lunarg) [reply scripts/genvp.py]: This is resolved in the following commit

**Pull request merged**